packages/flutter/lib/src/gestures/binding.dart

```diff
@@ -16,6 +17,15 @@ import 'events.dart';
 import 'hit_test.dart';
 import 'pointer_router.dart';
 import 'pointer_signal_resolver.dart';
+import 'resampler.dart';
+
+/// The default sampling offset.
+///
+/// Sampling offset is relative to presentation time. If we produce frames
+/// 16.667 ms before presentation and input rate is ~60hz, worst case latency
+/// is 33.334 ms. This however assumes zero latency from the input driver.
+/// 4.666 ms margin is added for this.
+const Duration _defaultSamplingOffset = Duration(milliseconds: -38);
```

Comment:
So should this number depend on the frame refresh rate? e.g. on 90Hz device like Pixel 4, should this be smaller?

Reply:
The ideal value typically depend on:

1. Sensor frequency.
2. Platform processing overhead. E.g. how long does it take for the event to arrive here.
3. The amount of input latency that an app is willing to add for smooth event processing.

Best way to find a good default values here are through experimentation. The value used by default for now has been proven to work well on Fuchsia systems where the sensor is ~60hz. Shorter duration will result in poor resampling results as it becomes more likely that we don't have events queued that can be used for resampling.

Reply:
We could reduce this to -21.333 and add refresh rate interval on top of it. Not sure that's much better though as the 21.333 value is still what needs to be determined through testing. We currently add 16.667 to this value here assuming that we start producing a frame for the currentSystemFrameTimeStamp at that offset. That's not the best assumption but making it "1000 / refreshRate" is also not good as that gives the appearance that we're always using the interval as frame scheduling offset.